Migrating away from Travis

ryanrath opened the pull request:
**NOTE: This PR must be merged before https://github.com/ubccr/xdmod/pull/1463** 


- As we're now using shippable for all of our testing needs we no longer need
  nvm to install a specific version of node as it's already pre-installed on
  the docker image that we're using with Shippable.
- I've had eslint & phpunit generate junit test artifacts so that they can be
  viewed via the Shippable interface.

ryanrath committed:
Removing NVM and use the native npm / node

Removing npm_initialized references in install.sh

Since we're no longer updating npm / installing nvm this code can be removed.

ryanrath committed:
Adding unit output for eslint & phpunit tests

Updated these two commands so that they are run as they were before if the
`$SHIPPABLE_BUILD_DIR` env variable is not present.

ryanrath committed:
Renaming `travis` directory to `scripts`

Since we're not going to be utilizing Travis anymore it behoves us to rename
this directory to minimize the wtf's / minute of future developers.

ryanrath committed:
Updated README to take into travis dir rename

ryanrath committed:
UPAdalksdjf

ryanrath committed:
Changed module_dir -> module_name

ryanrath committed:
Fixing copy/paste problem

plessbd reviewed (approved):
I wonder if we should just switch this to a v2 branch... but code is approved

ryanrath committed:
Updating README.md & shippable.yml

- Updated the Travis section of README.md along with its contents to now
  reference / be applicable to Shippable.
- Updated `scripts/template.yml` to be a template for shippable.yml instead of
travis.yml

ryanrath commented:
I just want to document that the https://github.com/ubccr/xdmod-qa/pull/14/commits/a2aa7246c5e4abd291d5a3153964c394e46a073f commit was made because the ultimate series of steps needed to utilize this repo w/ the new Shippable changes in both XDMoD & the various modules meant that it was easier to give control of these steps to the individual projects ( many of them were already performing these steps ).
